### PR TITLE
AssetSlice -> EntitySlice

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

Similar to the previous PR, but this time for AssetSlice. This lets us create and use slices of arbitrary EntityKeys. To help make this a bit simpler, I refactored the internal representation of an AssetSlice / EntitySlice to store a separate key/value so that the type system could independently understand the key type. I also removed some unused methods / code paths to aovid needing to refactor them . In particular, I removed the `time_windows` property of AssetSlice as it was unused outside of tests. We can always bring that back if we find a use for it.

## Changelog [New | Bug | Docs]

NOCHANGELOG

### CODE REVIEW

**Comment by OwenKephart** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23910?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#23910** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23910?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#23907** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23907?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23905** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23905?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23904** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23904?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23843** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23843?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#23842](https://github.com/dagster-io/dagster/pull/23842) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23842?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @OwenKephart and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py [hunk 1]
```diff
@@ -72,65 +59,35 @@ class TemporalContext(NamedTuple):
...
-    * Methods prefixed with `compute_` do potentially expensive work, like compute
-      partition mappings and query the instance.
-
-      We also use this prefix to indicate that they fully materialize partition sets
-      These can potentially be very expensive if the underlying partition set has
-      an in-memory representation that involves large time windows. I.e. if the
-      underlying PartitionsSubset in the AssetSubset is a
-      TimeWindowPartitionsSubset. Usage of these methods should be avoided if
-      possible if you are potentially dealing with slices with large time-based
-      partition windows.
```
Comment (schrockn) [inline python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py:123]: Is this docblock moved somewhere?

**Review by schrockn — CHANGES_REQUESTED** [review]
To discuss today. I believe you floated moving from "slice" to "subset" here which I'm sympathetic to.

**Comment by OwenKephart** [comment]
@schrockn I have a PR upstack which does the rename of EntitySlice -> EntitySubset

**Comment by OwenKephart** [comment]
### Merge activity

* **Sep 19, 12:04 PM EDT**: @OwenKephart started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/23910).

**Pull request merged**